extensions/ql-vscode/src/cli.ts

```diff
@@ -482,6 +497,20 @@ export class CodeQLCliServer implements Disposable {
     return await this.runJsonCodeQlCliCommand<QuerySetup>(['resolve', 'library-path'], subcommandArgs, 'Resolving library paths');
   }
 
+  /**
+   * Resolves the language for a query.
+   * @param queryUri The URI of the query
+   */
+  async resolveQueryByLanguage(workspaces: string[], queryUri: Uri): Promise<QueryInfoByLanguage> {
```

Comment:
Since you asked about tests:
- `extensions/ql-vscode/src/vscode-tests/cli-integration/run-cli.test.ts` is a good place to add integration tests for the `resolveQueryByLanguage` and `resolveLanguages` functions.
- You can use the existing tests as an example.
- `resolveLanguages` is probably easier to write a test for because there's no query as input. 
- `resolveQueryByLanguage` will require a test query: there are some examples in the `data` folders, but as far as I can tell none of them depend on an actual language. So you may need to add one, or save this test for later.